### DISCUSSION TITLE
usePermission for geolocation remain on 'prompt' after user turns on location services

### DISCUSSION THREAD

**Discussion by epeleg** [vueuse/vueuse#3874] [General]
I am posting this here and not as an issuse as I am unable to provide a reproduction environment.
maybe someone else would be able to do that or mabye it will just help someone else with the same issue.

Anyway,
I was using a combination of usePermission together with useGeolocation and encountered the following scenario:
I am using a PWA and when the phone owner had the location services turned off

my code initializes like this:

```
import { usePermission } from '@vueuse/core';
const geoPermissionState = usePermission('geolocation');
import { useGeolocation } from '@vueuse/core';
const { coords, resume,   pause } = useGeolocation({immediate:false,maximumAge:10*60*1000, timeout:27*1000});
```

`geoPermissionsState.value` was 'propmt' which is o.k..

Seeing that this was the situation I was calling resume() following some explaining to the user and him pressing a button.
This brought up a notification for the user prompitng him to turn on his location services. 
Even though the user did just that, the value of  `geoPermissionState.value` remained 'prompt'.
This is in contrast with the more standard scenario where location services are turned on and the user has not yet provided permission in which case the users consent to  share his location results in the `geoPermissionState.value` changing to 'granted' as expected.
I have than discovered that though  `geoPermissionState.value` remained 'prompt' -  The `coords.latitude` did start to provide the expected data - so I modified my logic to react not only to a change in the reported permissions but also to a change in the coords and was able to circumvent the issue.

**Comment by ilyaliao** [comment]
If the issue still exists, please create an issue. We would be glad to resolve it.

**Comment by epeleg** [comment]
to be honest, I don't know, this is from almost a year ago, and as I wrote there - i did find myself a workaround...
I can still say that the entire process is not very clear - and there are many "exceptions". 
for example - if my pwa is running inside an iphone than location permissions should be granted on two levels:
First for the safari application and only than to my pwa - and I am not sure that the two are programmatically distinguishable.
I also don't remember of no permissions are distinguishable from locations services being turned off.
I will try to create a chart of some sort to map the different scenarios.

**Comment by oscaroteromar** [comment]
Hey!

When trying to get whether the persmission has been granted to access the device's geolocation, the value returned by `usePermission('geolocation')`  remains as "prompt" even though the user has indeed granted the access.

I see this behaviour in iOS (18.4.1) and macOS (Sequoia 15.2), on different browsers.

However, I could test this scenario in other OS (Linux, Windows) and this is not a problem, the output of `usePermission('geolocation')` is correctly updated from "prompt" to "granted".

To try this, you can go to this [vueuse playground](https://playground.vueuse.org/?vueuse=13.3.0#eNp9VMtu2zAQ/BVCF7uAIAFtT4bSxi2CojkkRt2jLgy1lpVQJMGHY0PQv2dF6pmXLuLuzo6G4iybaKtUcnIQbaLMMF0pSwxYpwinorzKI2vy6EcuqlpJbUlDnIE/ILlk1FZSxF28A11XxmBIWnLQsiara2TESsqkhlUucsGkMMhsqXWGXC271qtyYlx9GcANYVLqwsTE16DY2piA1lLHRINxNcREUWTCr3rGma41smRp2A+qx8BCrTiyYERIVlQnv8ClQlUXDrhVhu16g9wFbjlI3ZCmGVS3bZaqsUvD8IcejRTY0DShdLu/v0uM1ZUoq8NlHZKE9NXuCdtC6ilFCGXMacoum76cDIl4jsIdVNYVMKKGxBIl8dtL2JBZ4Ch/xTYk3kVt3yh8VVh0HYEW+AtGcB8vMEYBFCPCR7N6O1tPDphy3gqb8CI/wzupwRhaAunzI3wiE47zMfjar9AuiOkOWAeH4BE/OGvR0teMV+wJz9l7zQ9D6N157z1Ty459RxpaPiIIpp0x7C3FmfqIIUuDS7N05t0oxonE+ThUZdIZD8fW26hzb60qDvpedQOAUzsaLI8o5/L51uesdsPxYs8R2NM7+Udz7nJ5tEPNoE8oeqyh5hJsKN/s7+CM67FYy8JxRH9S/AdGctdpDLBfThQoe4bzav/6Cwcd89/cnC2I7qIYhfrT8vg8wpvm9ydbn+R+S777vly0UfsCy3Ga8g==) and see the red "status" element value.

Another consideration about this, in Apple devices, if the location has been granted to a certain page, this change is not reflected in the `Privacy & Security > Location` browser's section. However, in other OS this permission granted appears there.

See a screen recording of all this. When I grant access to the location, this permission does not appear in the browser settings and the "prompt" value is not updated even though the coords are successfully obtained.

https://github.com/user-attachments/assets/5b7b564e-3652-4ab0-b8ea-fc96a5c2482d


As a conclusion, I guess it's not a bug related to vueuse but more about about iOS / macOS devices.
